[ML] Remove unnecessary version constraints

edsavage opened the pull request:
Remove checks on version numbers from the serialisation code that are no longer necessary in the 8.0.0 branch.

Labelling as non-issue as this should have no functional effect.

edsavage committed:
[ML] Remove unnecessary version constraints

Remove checks on version numbers from the serialisation code that are no longer necessary in the 8.0.0 branch

elasticmachine commented:
Pinging @elastic/ml-core (:ml)

droberts195 commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDataFrameAnalyticsStatsAction.java:
```diff
@@ -193,26 +192,13 @@ public Stats(StreamInput in) throws IOException {
                 state = DataFrameAnalyticsState.fromStream(in);
                 failureReason = in.readOptionalString();
-                if (in.getVersion().before(Version.V_7_4_0)) {
-                    progress = readProgressFromLegacy(state, in);
```
For this file it is probably worth leaving the BWC code in master for a bit longer.  It's very complex and would be a nasty shock for someone seeing it for the first time on a backport to 7.x.

droberts195 commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelAction.java:
```diff
@@ -72,18 +67,7 @@ public Request(StreamInput in) throws IOException {
-                } else {
-                    throw new IOException("Unexpected configuration type [" + oldConfig.getName() + "]");
-                }
-            }
```
As above, this is very complex and it would be useful for anyone changing it on the master branch to be aware the complexity exists in 7.x.  So it's probably best to leave this file untouched until closer to 8.0 feature freeze too.

droberts195 commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsState.java:
```diff
@@ -29,11 +28,7 @@ public static DataFrameAnalyticsState fromStream(StreamInput in) throws IOExcept
-            toWrite = STOPPED;
-        }
+
         out.writeEnum(toWrite);
```
There's no need to have a separate local variable now.

```suggestion
        out.writeEnum(this);
```

plus delete the line `DataFrameAnalyticsState toWrite = this;`

droberts195 commented on x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigTests.java:
```diff
@@ -468,24 +451,6 @@ public void testToXContent_GivenAnalysisWithRandomizeSeedAndVersionIsCurrent() t
         }
     }
 
-    public void testToXContent_GivenAnalysisWithRandomizeSeedAndVersionIsBeforeItWasIntroduced() throws IOException {
```
This needs to stay.  The 7.5 here is embedded in the config document, not the node version.  Unless we do something else 8.x _will_ still have to cope with configs created in 7.5 and before.

droberts195 commented on x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/ClassificationTests.java:
```diff
@@ -299,16 +299,6 @@ public void testGetExplicitlyMappedFields() {
             equalTo(Collections.singletonMap("results.feature_importance", MapUtils.featureImportanceMapping())));
     }
 
-    public void testToXContent_GivenVersionBeforeRandomizeSeedWasIntroduced() throws IOException {
```
As with outlier detection, the version here is in the config, not the current node version, so this should stay.

droberts195 commented on x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/RegressionTests.java:
```diff
@@ -221,17 +221,6 @@ public void testExtractJobIdFromStateDoc() {
         assertThat(Regression.extractJobIdFromStateDoc("noop"), is(nullValue()));
     }
 
-    public void testToXContent_GivenVersionBeforeRandomizeSeedWasIntroduced() throws IOException {
```
As with outlier detection, the version here is in the config, not the current node version, so this should stay.

droberts195 reviewed (commented):
Thanks for doing this.

I think there are just a couple of places where it would be useful for people working in the master branch to be aware of the BWC complexity in 7.x and also the data frame analytics config version looks like it's a JSON doc thing rather than a wire serialization thing, so will still need to be handled in 8.x (or avoided by some separate project).

edsavage committed:
[ML] Attending to code review comments

Reverting the more complex changes as these are best dealt with after
8.0.0 feature freeze.

edsavage committed:
Merge branch 'master' of github.com:elasticsearch/elasticsearch into remove_version_constraints

droberts195 commented on x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobNodeSelector.java:
```diff
@@ -82,10 +81,7 @@ public JobNodeSelector(ClusterState clusterState, String jobId, String taskName,
-        boolean allNodesHaveDynamicMaxWorkers = clusterState.getNodes().getMinNodeVersion().onOrAfter(Version.V_7_2_0);
-
-        // Try to allocate jobs according to memory usage, but if that's not possible (maybe due to a mixed version cluster or maybe
+       // Try to allocate jobs according to memory usage, but if that's not possible (maybe due to a mixed version cluster or maybe
```
nit: a space has accidentally been removed from this line

droberts195 reviewed (approved):
LGTM, although please add back the space that was accidentally removed from one line before merging

edsavage committed:
[ML] Further review comments

Reinstating accidentally removed space

edsavage commented:
retest

edsavage committed:
Merge branch 'master' of github.com:elasticsearch/elasticsearch into remove_version_constraints
